app/services/brapi/v1/germplasm_queries.rb

```diff
@@ -122,17 +139,23 @@ def execute_statement(sql, atts)
     # at least until the previous one has been deallocated
     Thread.exclusive do
       @connection.prepare('brapi_statement', sql)
-      
-      if( (atts != nil) && !(atts.empty?) )
-        results = @connection.exec_prepared("brapi_statement", atts)
-      else
-        results = @connection.exec_prepared("brapi_statement")
+      begin
+        if( (atts != nil) && !(atts.empty?) )
+          results = @connection.exec_prepared("brapi_statement", atts)
+        else
+          results = @connection.exec_prepared("brapi_statement")
+        end
+      rescue PG::Error => e
+        @connection.exec("ROLLBACK") 
+        results = nil
+        Rails.logger.warn { "Encountered an error executing a BrAPI germplasm-related query: #{e.message} #{e.backtrace.join("\n")}" }
```

Comment:
Thank you very much for your guidance. I've being doing some tests about this with studies/id call and I've created a new custom exception QueryError into new module 'Errors::Brapi' (app/errors is now included into autoload_paths).
If the new approach is correct I would extend it into all other calls.